Icon for nZombies

func-octo opened the pull request:
I have created an appropriate Icon for nZombies and was wondering if you were interested. Also I'd like to help contribute to this amazing, not so well-known project :)

Also, it would be nice if you could add a config option that lets us decide to have the normal random box or one that just gives us the random weapon without the lag-inducing weapon-model cycle. Any way to remove corpses of zombies after round? After about round 10, it starts lagging badly. Finally, the AddNotify method isn't working and the errors are lagging my game.

lolleko committed:
Merge remote-tracking branch 'Zet0rz/Master-Changes' into Master-Changes

lolleko committed:
Added new Round System

Mostly internal changes

Added the following shared hooks:
OnRoundWating
OnRoundInit
OnRoundPreperation
OnRoundStart
OnRoundEnd

Added the following server hooks:
OnPlayerReady
OnPlayerUnReady
OnPlayerDropIn
OnPlayerDropOut

Reworked round networking

Added some player meta functions in addition to the new features

Zet0rz committed:
Merge pull request #9 from lolleko/Master-Changes

Round rework

Zet0rz committed:
Door System Rewrite

Rewritten the Door System to be more modular and following the same
format as the new Round system. Also fixed a few bugs related to the
round system change. This may be buggy.

Should this version be unplayable, refer to the
Master-Changes_nzTableFallback branch which has all the same gameplay
features but in the old formatting as we rewrite Master-Changes.

lolleko committed:
Merge remote-tracking branch 'Zet0rz/Master-Changes' into Master-Changes

lolleko committed:
Merge remote-tracking branch 'Zet0rz/Master-Changes' into Master-Changes

lolleko committed:
Wepswitching

Added quick weapon switch via keys 1-3

Zet0rz committed:
Merge pull request #10 from lolleko/Master-Changes

Wep swithing

Zet0rz committed:
A Sharp Update!

New stuff and bug fixes!
- Added Crowbar Quickknife (Use with V)
- DMG_CLUB damage types now give 130 points on kill
- Added Special Weapons system to later allow for grenades and other
special weapons
- Fixed mismatches with the new and old door and round system around the
code
- Fixed doors not properly working
- You can now no longer ready up when no navmesh is loaded and crash the
game
- Some nicer lua refresh safeties
- CleanUpMap now also removes decals client-side
- Fixed sync with the round number
- Fixed a few typos causing bugs
- Added all FAS2 weapons as PaP names

lolleko committed:
Merge remote-tracking branch 'Zet0rz/Master-Changes' into Master-Changes

lolleko committed:
Spectator module

lolleko committed:
Update sv_player.lua

added missing local(again :D)

lolleko committed:
fixed chat commands

lolleko committed:
replaced some function calls

lolleko committed:
Update sv_override.lua

lolleko committed:
Improved zombies prop pushing behaviour

lolleko committed:
Another replaced legacy function call

lolleko committed:
fixed typo

Zet0rz committed:
Merge pull request #11 from lolleko/Master-Changes

Spectator module + Prop Pushing behaviour

Zet0rz committed:
Restructure

Fixed a few bugs and desyncs related to the new door system, as well as
restructured the entire gamemode into an addon format.

Zet0rz committed:
Update README.md

lolleko committed:
Merge remote-tracking branch 'refs/remotes/Zet0rz/Master-Changes' into Master-Changes

lolleko committed:
Bugfixes

Changed the Random Box module to the new format
Improved complexity of GetAllPlayingAndAlive from O(2n) to O(n)
Fixed various bugs related to the new spectator system
Fixed main menu alignment for Windows
Improved zombie jump mechanics

Zet0rz committed:
Merge pull request #12 from lolleko/Master-Changes

Bugfixes

Zet0rz committed:
Fog Approaches

Added autogenerated speed curves - this will save caching memory
Added fog and fading fog functions. Use the fog editors to set them
based on your config.
Fixed a few misc bugs.

Zet0rz committed:
Special Rounds Approaches

A few new features and a pretty funny bug fix.
- Added Special Rounds to run on intervals
- The last zombie in a special round always drops max ammo
- Added new sounds to the special rounds
- Special Rounds now always follow one EnemyTypes table, but you can
overwrite it with a normal EnemyTypes one

- Fixed being able to reload more shots into your mag than you have
reserve ammo with Speed Cola

lolleko committed:
Merge remote-tracking branch 'refs/remotes/Zet0rz/Master-Changes' into Master-Changes

Zet0rz committed:
Fragtastic!

A lil' update featuring some cool things along with a feature requested.
- Added Frag Grenades. Use them by binding a key to "+grenade1"
- Added ability for Special Weapons to have On Max Ammo functions
- Weapons now get a 50% bigger mag on PaP
- PaP names now draw when trying to pick up a newly-pap'd weapon from
the machine
- A whitelist has been added allowing you to set prefixes for which
weapons will be loaded (Map Settings will always overwrite this unless
changed)
- Added an option to ignore Map Setting weapon lists in the config (and
then use whitelist/blacklist)
- Changed Blacklist table to be of O(1) complexity when checking

Zet0rz committed:
Removed Legacy Tools

They have been removed now. Most of them won't work anyway. The
interfaces still exist, but to be used when right clicking from the
Context Menu.

Zet0rz committed:
Quickly removed Special

Quickly removed Special status for Round 1

Zet0rz commented:
I can't guarantee we'll use the icon, but I'm happy to have a look :)
And feel free to contribute, but give a good explanation of your changes, and please don't create a pull request trying to merge an entire branch into another xD

As to your feedback, the weapon scrolling shouldn't really cause lag, but I can make it just give you the weapon from the start with a convar. A console command to remove corpses or a max corpse limit could also be something we should look at. And finally, you'll have to give me the full console error for me to fix that AddNotify thing.

lolleko committed:
Merge remote-tracking branch 'refs/remotes/Zet0rz/Master-Changes' into Master-Changes

lolleko committed:
Zombie Stuck prevention and random bugfixes

lolleko committed:
Added nmob of the dead models

Models by half-dead http://steamcommunity.com/id/Half_Dead

LICENSE?

Characters and zombies from Call of Duty Black Ops 2 Uprising DLC. Has
bodygroups, minor faceposing, red tinted screenshots and lots of barbed
wire!

These are not NPC's but they ARE rigged to valvebiped so if you want to
make them NPC's go right ahead as long as you give credit to me and
Treyarch!

New in V.2: added all player characters from MotD

Zet0rz committed:
Merge pull request #15 from lolleko/Master-Changes

zombie anti stuck + Mob of the Dead Models
